fenicsadapter/fenicsadapter.py

```diff
@@ -259,501 +38,308 @@ def __init__(self, adapter_config_filename='precice-adapter-config.json',
         self._interface = precice.Interface(self._solver_name, self._config.get_config_file_name(), 0, 1)
         self._dimensions = self._interface.get_dimensions()
 
-        self._coupling_subdomain = None  # initialized later
-        self._mesh_fenics = None  # initialized later
-        self._coupling_bc_expression = None  # initialized later
+        # FEniCS related quantities
         self._fenics_dimensions = None  # initialized later
```

Comment:
Add comment: Initialized in `initialize()`. Even nicer: Entirely get rid of this member and determine the dimension (from input data?) when needed. But if this is too troublesome, leave it as it is.

Reply:
See also my comment above on merging object creation via `__init__`, `set_interpolation_type(...)` and `initialize(...)` above.